miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -1075,11 +1079,16 @@ public ValidationResult validateDeletion(Sample object) {
         result.addError(new ValidationError(object.getName() + " has " + childCount + " child sample" + (childCount > 1 ? "s" : "")));
       }
     }
-    if (object.getLibraries() != null && !object.getLibraries().isEmpty()) {
-      result.addError(new ValidationError(object.getName() + " has " + object.getLibraries().size() + " librar"
-          + (object.getLibraries().size() > 1 ? "ies" : "y")));
+    try {
+    final List<Library> libraries = libraryService.listBySampleId(object.getId());
+      if (libraries.isEmpty()) {
+        result.addError(new ValidationError(object.getName() + " has " + libraries.size() + " librar"
+            + (libraries.size() > 1 ? "ies" : "y")));
+      }
+    } catch (IOException e) {
+       log.error("Failed to fetch libraries for " + object.getName(), e);
+       result.addError(new ValidationError("Failed to check for libraries."));
     }
```

Comment:
* Condition should be negated, which will fix the failing test too
* Use `ValidationError#forDeletionUsage` to construct standard error message
* Use `Pluralizer#libraries` for `referrer` parameter of `forDeletionUsage`
* throw the `IOException` (already declared in `DeleterService#validateDeletion`) as that shouldn't be reported as a validation error)